client/App/Containers/Data/DataScreen.js

```diff
@@ -3,28 +3,51 @@ import { TouchableOpacity, Text, View, ActivityIndicator, Image } from 'react-na
 import { connect } from 'react-redux'
 import PropTypes from 'prop-types'
 import WifiActions from 'App/Stores/Wifi/Actions'
-import Style from './WifiListScreenStyle'
+import GpsActions from 'App/Stores/Gps/Actions'
+import Style from './DataScreenStyle'
 import { Helpers, Images, Metrics } from 'App/Theme'
 
-class WifiListScreen extends React.Component {
+class DataScreen extends React.Component {
   componentDidMount() {
-    this._fetchWifiList();
+    this._fetchData();
   }
 
-  _fetchWifiList = () => {
-    this.props.fetchWifiList()
+  _fetchData = () => {
+    this.props.fetchWifiList();
+    this.props.fetchGpsLocation();
   }
   
   componentDidUpdate(prevProps) {
-    const {wifiList} = this.props;
+    const {wifiList} = this.props.wifiList;
     const {wifiList: prevWifiList} = prevProps;
     
     if (JSON.stringify(wifiList) != JSON.stringify(prevWifiList)) {
       console.log('wifiList ?', wifiList);
+    } else {
+      console.log('Wifi did not change :-(');
     }
   }
 
   render() {
+
+    // Handle gpsLocation being undefined
+    let gpsData;
+    if(this.props.gpsLocation){
+      gpsData = 
+          <Text>
+            {"\n"}     
+            accuracy: {this.props.gpsLocation.accuracy}, {"\n"}
```

Comment:
lets make it more readable:
`const { gpsLocation} = this.props;`
then - you can:
`gpsLocation.accuarcy`...